### ISSUE TITLE
warnings with clang on NetBSD 9.99.4/amd64

### ISSUE THREAD

**Issue by ng-0** [resurrecting-open-source-projects/scrot#26]
Hi,

while evaluating this as an update for graphics/scrot in pkgsrc, I've got these warnings (on NetBSD 9.99.4 with clang + llvm 8). Fixes should be simple.

```
Making all in src
/usr/bin/make  all-am
clang -DHAVE_CONFIG_H -I.  -g -O3 -Wall -I/usr/X11R6/include   -I/usr/pkg/include -I/usr/pkg/include -I.  -DPREFIX=\""/usr/pkg"\" -I/usr/pkg/include -I/usr/pkg/include/giblib -I/usr/pkg/include  -I/usr/pkg/include -I/usr/include -I/usr/X11R7/include/freetype2 -I/usr/X11R7/include  -O2 -I/usr/pkg/include -I/usr/include -I/usr/X11R7/include/freetype2 -I/usr/X11R7/include -MT main.o -MD -MP -MF .deps/main.Tpo -c -o main.o main.c
mv -f .deps/main.Tpo .deps/main.Po
clang -DHAVE_CONFIG_H -I.  -g -O3 -Wall -I/usr/X11R6/include   -I/usr/pkg/include -I/usr/pkg/include -I.  -DPREFIX=\""/usr/pkg"\" -I/usr/pkg/include -I/usr/pkg/include/giblib -I/usr/pkg/include  -I/usr/pkg/include -I/usr/include -I/usr/X11R7/include/freetype2 -I/usr/X11R7/include  -O2 -I/usr/pkg/include -I/usr/include -I/usr/X11R7/include/freetype2 -I/usr/X11R7/include -MT getopt.o -MD -MP -MF .deps/getopt.Tpo -c -o getopt.o getopt.c
getopt.c:571:30: warning: implicitly declaring library function 'strcmp' with
      type 'int (const char *, const char *)' [-Wimplicit-function-declaration]
      if (optind != argc && !strcmp(argv[optind], "--"))
                             ^
getopt.c:571:30: note: include the header <string.h> or explicitly provide a
      declaration for 'strcmp'
getopt.c:649:15: warning: implicitly declaring library function 'strncmp' with
      type 'int (const char *, const char *, unsigned long)'
      [-Wimplicit-function-declaration]
         if (!strncmp(p->name, nextchar, nameend - nextchar))
              ^
getopt.c:649:15: note: include the header <string.h> or explicitly provide a
      declaration for 'strncmp'
getopt.c:652:32: warning: implicitly declaring library function 'strlen' with
      type 'unsigned long (const char *)' [-Wimplicit-function-declaration]
                (unsigned int) strlen(p->name))
                               ^
getopt.c:652:32: note: include the header <string.h> or explicitly provide a
      declaration for 'strlen'
getopt.c:701:19: warning: add explicit braces to avoid dangling else
      [-Wdangling-else]
                  else
                  ^
4 warnings generated.
```

**Comment by ng-0** [comment]
Then again, it's in getopt.c from GNU getopt. 
Is the right patch instead on our side?

**Comment by ng-0** [comment]
For 1.1.1 release this patch by fox@ applies:

http://cvsweb.netbsd.org/bsdweb.cgi/pkgsrc/graphics/scrot/patches/patch-src_getopt.c?rev=1.1&content-type=text/x-cvsweb-markup&only_with_tag=MAIN

**Comment by eribertomota** [comment]
Hi @ng-0,

Thanks for your patch. 

@daltomi, it looks good for me. What you think about the patch? I can apply it.

**Comment by daltomi** [comment]
Hi,
Yes, I think it's fine.

**Comment by eribertomota** [comment]
I will commit soon. (today)

Em dom, 18 de ago de 2019 às 18:33, Daniel T. Borelli <
notifications@github.com> escreveu:

**Comment by eribertomota** [comment]
Reflecting about... @ng-0, can you submit a push request to register you as collaborator?

**Comment by ng-0** [comment]
I'm a NetBSD developer, but it's not my patch. I can ask fox@ if they are on github and want to handle this or if I can do it.

**Comment by eribertomota** [comment]
Ok, waiting. Thanks!

Em dom, 18 de ago de 2019 às 19:14, ng0 <notifications@github.com> escreveu:

**Comment by fraggerfox** [comment]
> 
> 
> Reflecting about... @ng-0, can you submit a push request to register you as collaborator?

Hello, would you like me to fork the repository and create a pull request with the above patch applied?

**Comment by eribertomota** [comment]
> > Reflecting about... @ng-0, can you submit a push request to register you as collaborator?
> 
> Hello, would you like me to fork the repository and create a pull request with the above patch applied?

If you are the fox@, yes!

**Comment by fraggerfox** [comment]
> 
> 
> > > Reflecting about... @ng-0, can you submit a push request to register you as collaborator?
> > 
> > 
> > Hello, would you like me to fork the repository and create a pull request with the above patch applied?
> 
> If you are the fox@, yes!

Yes I am! Let me do that.

**Comment by eribertomota** [comment]
Go ahead! Thanks!

Em dom, 18 de ago de 2019 às 19:32, Santhosh Raju <notifications@github.com>
escreveu:

**Comment by fraggerfox** [comment]
@eribertomota Let me know if the pull request is sufficient. Happy to make changes if needed.

**Comment by eribertomota** [comment]
Fixed by PR #27
